hazelcast-sql-core/src/test/java/com/hazelcast/sql/impl/expression/predicate/InPredicateIntegrationTest.java

```diff
@@ -0,0 +1,144 @@
+        checkValues(sqlQuery("NOT IN " + longList), SqlColumnType.INTEGER, new Integer[]{3});
+    }
+
+    @Test
+    public void inPredicateWithRawNullTest() {
+        putAll(0, 1, 2);
+        checkValues(sqlQuery("IN (NULL, 0, 1)"), SqlColumnType.INTEGER, new Integer[]{0, 1});
+        checkValues(sqlQuery("IN (0, NULL, NULL, NULL, 2)"), SqlColumnType.INTEGER, new Integer[]{2, 0});
+        checkValues(sqlQuery("NOT IN (0, NULL, NULL, NULL, 2)"), SqlColumnType.INTEGER, new Integer[]{1});
+        checkValues(sqlQuery("NOT IN (NULL, 2, 1)"), SqlColumnType.INTEGER, new Integer[]{0});
+    }
+
+    @Test
+    public void inPredicateWithSubQueryTest() {
+        putAll(1, 2);
+        checkThrows(sqlQuery("IN (SELECT __key FROM map)"), HazelcastSqlException.class);
+        checkThrows(sqlQuery("NOT IN (SELECT __key FROM map)"), HazelcastSqlException.class);
+    }
+
+    @Test
+    public void inPredicateNumberTypeTest() {
+        putAll(0, 1);
+        checkValues(sqlQuery("IN (0, '1', 2)"), SqlColumnType.INTEGER, new Integer[]{0, 1});
+        checkValues(sqlQuery("IN ('0', 1, '2')"), SqlColumnType.INTEGER, new Integer[]{0, 1});
+        checkValues(sqlQuery("IN (CAST ('0' AS INTEGER), CAST ('1' AS INTEGER), 2)"), SqlColumnType.INTEGER, new Integer[]{0, 1});
+        checkValues(sqlQuery("NOT IN (CAST ('0' AS INTEGER), CAST ('1' AS INTEGER), 2)"), SqlColumnType.INTEGER, new Integer[]{});
+    }
+
+    @Test
+    public void inPredicateStringTypeTest() {
+        putAll("abc", "bac", "cba");
+        checkValues(sqlQuery("IN ('abc', 'cba')"), SqlColumnType.VARCHAR, new String[]{"cba", "abc"});
+        checkValues(sqlQuery("IN ('abc', 'cba', 'bac')"), SqlColumnType.VARCHAR, new String[]{"cba", "abc", "bac"});
+        checkValues(sqlQuery("NOT IN ('abc', '1', 'cba')"), SqlColumnType.VARCHAR, new String[]{"bac"});
+    }
+
+    @Test
+    public void inPredicateDatesTest() {
+        LocalDate date1 = LocalDate.of(1970, 1, 1);
+        LocalDate date2 = LocalDate.of(1970, 1, 3);
+        putAll(date1, date2);
+        checkValues(sqlQuery("IN (CAST ('1970-01-01' AS DATE), CAST ('1970-01-03' AS DATE))"), SqlColumnType.DATE, new LocalDate[]{date1, date2});
+        checkValues(sqlQuery("IN (CAST ('1970-01-01' AS DATE), CAST ('1970-01-02' AS DATE))"), SqlColumnType.DATE, new LocalDate[]{date1});
+        checkValues(sqlQuery("NOT IN (CAST ('1970-01-01' AS DATE), CAST ('1970-01-02' AS DATE))"), SqlColumnType.DATE, new LocalDate[]{date2});
+    }
+
+    @Test
+    public void inPredicateWithDifferentTypesWouldFailsTest() {
+        putAll(0, 1, "abc", LocalDate.of(1970, 1, 1));
+        checkThrows(sqlQuery("IN ('1970-01-01', 1, '2')"), HazelcastSqlException.class);
+        checkThrows(sqlQuery("IN (0, 1, 'bac')"), HazelcastSqlException.class);
+        checkThrows(sqlQuery(sqlQuery("IN (CAST ('1970-01-01' AS DATE), 1, CAST ('1970-01-02' AS DATE))")), HazelcastSqlException.class);
```

Comment:
Could you please add similar tests but map values of the same type?
I think you will need to test it when map types are:
* numeric
* string
* temporal